### ISSUE TITLE
Pseudo parameters no longer work in custom variables or resources

### ISSUE THREAD

**Issue by adsurg** [svdgraaf/serverless-pseudo-parameters#10]
The earlier evaluation means that "#{AWS::StackName}-lock" is no longer being substituted.  This is working in v1.3.2.

```
resources:
  Resources:
    lambdaLockDb:
      Type: 'AWS::DynamoDB::Table'
      ...
      TableName: "#{AWS::StackName}-lock"
```

```
provider:
  name: aws
  ...
  environment:
    LOCK_TABLE: ${self:custom.lockTableName}
custom:
  lockTableName: "#{AWS::StackName}-lock"
```

**Comment by svdgraaf** [comment]
Ok, yeah, that's not good. Perhaps I should start adding tests ;)
I'll see if I can fix this, not sure if I have time today though.

**Comment by adsurg** [comment]
That was my reaction :-).  I thought I'd have a look at fixing this, but the lack of tests made me fearful of just breaking your previous issue.  Cheers for replying so quickly.

**Comment by komushi** [comment]
Had the same issue now with #{AWS::Region} under resource.
I was using 1.1.5 and it was working but now even if I set the version to a fixed 1.1.5 it still fails.

**Comment by adsurg** [comment]
Version 1.3.2 is the same as v1.4, but without this change.  We've stopped at that version for now.

**Comment by komushi** [comment]
Thanks. Will fix it to 1.3.2 for now

**Comment by lorengordon** [comment]
Submitted #11 with a patch for this issue

**Comment by svdgraaf** [comment]
I just released 1.4.1, can you check whether that fixes your issues?

**Comment by svdgraaf** [comment]
Ok, closing this one :)
